service/common/src/main/java/org/apache/polaris/service/catalog/policy/PolicyCatalog.java

```diff
@@ -249,27 +246,246 @@ public Policy updatePolicy(
     return constructPolicy(newPolicyEntity);
   }
 
-  public boolean dropPolicy(PolicyIdentifier policyIdentifier, boolean detachAll) {
+  public void dropPolicy(PolicyIdentifier policyIdentifier, boolean detachAll) {
     // TODO: Implement detachAll when we support attach/detach policy
-    PolarisResolvedPathWrapper resolvedEntities =
-        resolvedEntityView.getPassthroughResolvedPath(
-            policyIdentifier, PolarisEntityType.POLICY, PolarisEntitySubType.NULL_SUBTYPE);
-    if (resolvedEntities == null) {
-      throw new NoSuchPolicyException(String.format("Policy does not exist: %s", policyIdentifier));
-    }
+    var resolvedPolicyPath = getResolvedPathWrapper(policyIdentifier);
+    var catalogPath = resolvedPolicyPath.getRawParentPath();
+    var policyEntity = resolvedPolicyPath.getRawLeafEntity();
 
-    List<PolarisEntity> catalogPath = resolvedEntities.getRawParentPath();
-    PolarisEntity leafEntity = resolvedEntities.getRawLeafEntity();
-
-    DropEntityResult dropEntityResult =
+    var result =
         metaStoreManager.dropEntityIfExists(
             callContext.getPolarisCallContext(),
             PolarisEntity.toCoreList(catalogPath),
-            leafEntity,
+            policyEntity,
             Map.of(),
             false);
 
-    return dropEntityResult.isSuccess();
+    if (!result.isSuccess()) {
```

Comment:
throwing an exception here looks incorrect to me, or at least inconsistent with IcebergCatalog and GenericTableCatalog. From the user POV, the return of this method doesn't matter but rather the return code (and content) of the REST API matters.

Reply:
> From the user POV, the return of this method doesn't matter but rather the return code (and content) of the REST API matters.

Not sure I understand completely. The http return code is controlled by the exception mapper, like `IcebergExceptionMapper` and `PolarisExceptionMapper`.  

> throwing an exception here looks incorrect to me, or at least inconsistent with IcebergCatalog and GenericTableCatalog

It is inconsistent with a method like `public boolean dropTable(TableIdentifier tableIdentifier, boolean purge)` in `IcebergCatalog`, but I believe it's the right thing to do. The `dropTable` method needs to return a boolean only because it is inherited from the method in a class from Iceberg lib. I didn't see any reason we need a boolean return. Or am I missing something?

Reply:
> Not sure I understand completely. The http return code is controlled by the exception mapper, like IcebergExceptionMapper and PolarisExceptionMapper.

This is true, but this method is not the final touchpoint between the business logic and the REST logic. From what I understand, there will be another layer on top of this like `IcebergCatalogAdapter`, right?

> I didn't see any reason we need a boolean return. Or am I missing something?

We fundamentally have a boolean outcome here -- the drop either succeeds or it does not. We could use an exception for flow control here, but if we can achieve the same outcome with a boolean shouldn't we prefer that?

Reply:
Added back boolean return per offline discussion.